spec/aci.md

```diff
@@ -239,6 +240,7 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
         * **port** (integer, required) port number that will be used; see also **count**. Must be >=1 and <=65535.
         * **count** (integer, optional, defaults to 1 if unset) specifies a range of ports, starting with "port" and ending with "port" + "count" - 1. Must be >=1.
         * **socketActivated** (boolean, optional, defaults to "false" if unsupplied) if set to true, the application expects to be [socket activated](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) on these ports. To perform socket activation, the ACE MUST pass file descriptors using the [socket activation protocol](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) that are listening on these ports when starting this app. If multiple apps in the same pod are using socket activation then the ACE must match the sockets to the correct apps using getsockopt() and getsockname().
+    * **supportNotify** (boolean, optional, defaults to "false" if unset) if set to true, the application should use [sd_notify()](https://www.freedesktop.org/software/systemd/man/sd_notify.html) to signal when it is ready. Setting this parameter to true for an application that does not send a ready signal with `sd_notify()` may result in misbehave.
```

Comment:
On Friday, June 17, 2016, Euan Kemp <notifications@github.com> wrote:

> In spec/aci.md
> <https://github.com/appc/spec/pull/626#discussion_r67568121>:
>
> > @@ -239,6 +240,7 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
> >          * **port** (integer, required) port number that will be used; see also **count**. Must be >=1 and <=65535.
> >          * **count** (integer, optional, defaults to 1 if unset) specifies a range of ports, starting with "port" and ending with "port" + "count" - 1. Must be >=1.
> >          * **socketActivated** (boolean, optional, defaults to "false" if unsupplied) if set to true, the application expects to be [socket activated](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) on these ports. To perform socket activation, the ACE MUST pass file descriptors using the [socket activation protocol](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) that are listening on these ports when starting this app. If multiple apps in the same pod are using socket activation then the ACE must match the sockets to the correct apps using getsockopt() and getsockname().
> > +    * **supportNotify** (boolean, optional, defaults to "false" if unset) if set to true, the application should use [sd_notify()](https://www.freedesktop.org/software/systemd/man/sd_notify.html) to signal when it is ready. Setting this parameter to true for an application that does not send a ready signal with `sd_notify()` may result in misbehave.
>
> perhaps it should be rephrased as "if set to true the application MUST use
> sd_notify" to more clearly express that the alternate behaviour is not well
> defined.
>
This sound good.

> I think the second case can happen more easily. For example, with
> stage1-fly there is not always a systemd to notify. Should stage1-fly
> refuse to run an application with supportsNotify=true, or should
> stage1-fly ignore it and the application be able to detect that it's not
> running under systemd and run normally?
>
I don't k ow how stage1-fly works, but for the moment the manly reason to
support notification is to have a a clear signal when the service in
the pod is ready. This allow to sort out dependencies between pods not
easily.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/appc/spec/pull/626/files/79ff3e69f02267582f5253175ac2cb01264a3bfa#r67568121>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGRc7zXAzEcr9FYT2I2E6ydhFbN1RJzVks5qMv7UgaJpZM4I2aiJ>
> .
>